x-pack/plugins/alerting/server/task_runner/inject_action_params.ts

```diff
@@ -9,27 +9,29 @@ import { i18n } from '@kbn/i18n';
 import { RuleActionParams } from '../types';
 
 export interface InjectActionParamsOpts {
-  ruleId: string;
-  spaceId: string | undefined;
   actionTypeId: string;
   actionParams: RuleActionParams;
+  ruleUrl?: string;
 }
 
 export function injectActionParams({
-  ruleId,
-  spaceId,
   actionTypeId,
   actionParams,
+  ruleUrl = '',
 }: InjectActionParamsOpts) {
   // Inject kibanaFooterLink if action type is email. This is used by the email action type
   // to inject a "View alert in Kibana" with a URL in the email's footer.
   if (actionTypeId === '.email') {
-    const spacePrefix =
-      spaceId && spaceId.length > 0 && spaceId !== 'default' ? `/s/${spaceId}` : '';
+    let path;
+    try {
+      path = new URL(ruleUrl).pathname;
+    } catch (e) {
+      path = '';
```

Comment:
In the email connector: https://github.com/elastic/kibana/blob/329cc6add68360a94de57b46338171d6f3869b46/x-pack/plugins/stack_connectors/server/connector_types/email/index.ts#L411-L418

it does this check `link: ${publicBaseUrl}${kibanaFooterLink.path === '/' ? '' : kibanaFooterLink.path}`

so should we set path to `/` in this error scenario?

Reply:
Yes i saw it but `"/"` becomes `""` anyway, and if we use it somewhere else without checking the trailing `"/"` the  full URL becomes broken.